modules/apps/portal/portal-aop-api/src/main/java/com/liferay/portal/aop/AopService.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.aop;
+
+/**
+ * Interface used to declare a service should be intercepted and proxied by an
+ * AOP and re-registered.
+ *
+ * <p>
+ * It is important that services required to be intercepted are only registered
+ * as a <code>AopService</code> so service listeners do not see the service
+ * before it is proxied.
+ * </p>
+ *
+ * @author Preston Crary
+ */
+public interface AopService {
+
+	/**
+	 * The interfaces under which to register this AopService.
+	 *
+	 * If null or empty, the service types for this AopService are all the
+	 * <i>directly</i> implemented interfaces of the AopService.
```

Comment:
@Preston-Crary What is a service type that is a directly implemented interface of AopService?

Do you mean a service type that directly implements AopService? In other words, are you saying service types that directly implement AopService aren't returned?

Reply:
The service proxy is registered into OSGi as a list of services, interface class names.

If the method is not overridden, the service proxy's list of services are the directly implemented interfaces excluding AopService.

There are several constraints on the service list:
1. It cannot include AopService.
2. It can only contain interfaces.
3. The interfaces must be implemented by the AopService implementation
4. It should be unchanging.
5. The AopService implementation must implement something else besides AopService to be valid.

Hope that answers your questions.